02-microarray/dimension-reduction_microarray_01_pca.Rmd

```diff
@@ -0,0 +1,312 @@
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE37382.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE37382.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene id column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+```
+
+
+<!--ALL THE ANALYSIS STEPS HERE -->
+
+Now we are going to use a combination of functions from base R and the `ggplot2` package to perform and visualize the results of the Principal Component Analysis (PCA) dimension reduction technique on our medulloblastoma samples.
+
+## Perform Principal Components Analysis
+
+In this code chunk, we are going to perform Principal Component Analysis (PCA) on our data and create a data frame using the PCA scores and the variables from our metadata that we are going to use to annotate our plot later.
+We are using the base R [`prcomp()`](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/prcomp) function to perform Principal Component Analysis (PCA) here [@prcomp-function].
+
+```{r}
+# Perform Principal Component Analysis (PCA)
+pca <- prcomp(t(df)) # We have to transpose our data frame to meet the `prcomp()` function requirements
+
+# Make into data frame for plotting PC1 and PC2 with `ggplot2`
+pca_df <- data.frame(pca$x[, 1:2]) %>%
+  # Turn samples_ids stored as rownames into column
+  tibble::rownames_to_column("refinebio_accession_code") %>%
+  # Bring only the variables that we want from the metadata into this data frame -- here we are going to join by `refinebio_accession_code` values
+  dplyr::inner_join(dplyr::select(
+    metadata, refinebio_accession_code,
+    histology, subgroup
+  ),
+  by = "refinebio_accession_code"
+  )
+```
+
+Let's plot the PC scores for the first two principal components.
+
+```{r}
+# Make a scatterplot using `ggplot2` functionality
+pca_plot <- ggplot(
+  pca_df,
+  aes(
+    x = PC1,
+    y = PC2
+  )
+) +
+  geom_point() + # This tells R that we want a scatterplot
+  theme_classic() # This tells R to return a classic-looking plot with no gridlines
+
+# Print out plot here
+pca_plot
+```
+
+Now, let's label the data points based on their genotype subgroup since this is central to the subgroup specific based hypothesis in the [original paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3683624/#SD1) [@Northcott2012].
```

Comment:
More general question - are we including hyperlinks _and_ citations?

Reply:
We have been doing so thus far, and this is what I gather from the [Citing sources in text section of `CONTRIBUTING.md`](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#citing-sources-in-text).
The `template.Rmd` file also instructs to use this format for the further resources section so for consistency purposes, I have been including the hyperlinks and citations.
Do you think this format should be adapted?

Reply:
That is not necessarily my read of that section of contributing for what we would do outside of the further resources section. I don't have strong feelings, but it does seem like maintaining two things (link to the paper and the citation) could be kind of a pain. Question for @cansavvy when she gets back.

Reply:
About the section in `Citing sources` in CONTRIBUTING.md. Here's what I'm assuming might have been the statement in question: 

> Note that citations do not have links, so links to helpful docs and vignettes need to be added separately.

I didn't intend for it to mean that everything that is cited must also have a link to something, but more that IF there is something helpful we want to point out, to know that the citations themselves don't have links, so adding a citation won't add a link. If this statement was more confusing than helpful, we can definitely reframe or remove it. 

I would also say that if something is in the `Further resources` section, in _most_ instances I don't think we need to link to it again in another place in the notebook. (I say most, because I could see instances where may there's a particular section of something that addresses a particular topic).